dumpling/export/sql.go

```diff
@@ -603,12 +605,38 @@ func GetTiDBDDLIDs(tctx *tcontext.Context, db *sql.DB) ([]string, error) {
 	return ddlIDs, errors.Annotatef(err, "sql: %s", query)
 }
 
+// getTiDBConfig gets tidb config from TiDB server
+// @@tidb_config details doc https://docs.pingcap.com/tidb/stable/system-variables#tidb_config
+// this variable exists at least from v2.0.0, so this works in most existing tidb instances
+func getTiDBConfig(db *sql.Conn) (dbconfig.Config, error) {
+	const query = "SELECT @@tidb_config;"
+	var (
+		tidbConfig      dbconfig.Config
+		tidbConfigBytes []byte
+	)
+	row := db.QueryRowContext(context.Background(), query)
+	err := row.Scan(&tidbConfigBytes)
+	if err != nil {
+		return tidbConfig, errors.Annotatef(err, "sql: %s", query)
+	}
+	err = json.Unmarshal(tidbConfigBytes, &tidbConfig)
```

Comment:
Is tidbConfig compatible across major versions? 🤔 Since we need to be compatible within at least 2 major versions.

Reply:
AFAIK `@@tidb_config` isn't available in SEM 🤔

Reply:
> Is tidbConfig compatible across major versions? 🤔 Since we need to be compatible within at least 2 major versions.
`@@tidb_config` exists at least from v2.0.0, so this works in most existing tidb instances.
> AFAIK @@tidb_config isn't available in SEM 🤔
It doesn't matter, we just reject users to use lock consistency in that case. In most cases `snapshot` consistency is better.

Reply:
@kennytm @glorv PTAL again